### PR TITLE
Allow increasing main storage size (sdcard is separate storage)

### PR DESCRIPTION
**Pull request by ViliusSutkus89**
Android has two drives - main storage for apps and external storage for downloads and user files.
`sdcard-path-or-size` controls the second - external storage.

# The Problem

Older devices default to 800M main storage. After the system apps, there is 309M left.
Most today's apps take more than that. Debug versions are not minimized, instrumented tests take extra space. Long story short - my app cannot fit. I get `INSTALL_FAILED_INSUFFICIENT_STORAGE` and related errors. Issue #184 is caused by this.

800M

# The PR

There is no option in `avdcreate`, but `/Users/runner/.android/avd/${avdName}.avd/config.ini` has the config option we want: `disk.dataPartition.size=800M`

This PR would allow appending a new `disk.dataPartition.size` size, when supplied as action argument named `disk-size`.

# The Workaround

Until this or similar PR is merged, I have a workaround:

```yaml
- name: Create AVD
  uses: reactivecircus/android-emulator-runner@v2
  with:
    api-level: 21
    arch: x86_64
    target: google_apis
    script: echo "Generated AVD snapshot."

- name: increate storage size for emulator
  run: echo "disk.dataPartition.size=2G" >> /Users/runner/.android/avd/test.avd/config.ini

- name: Run tests
  uses: reactivecircus/android-emulator-runner@v2
  with:
    api-level: 21
    arch: x86_64
    target: google_apis
    force-avd-creation: false
    script: ./gradlew connectedCheck
```

### CODE REVIEW

**Comment by ViliusSutkus89** [comment]
As an alternative, we could parse config.ini and increase this size for low storage devices without any user supplied input.

File: src/input-validator.ts [hunk 1]
```diff
@@ -69,3 +69,20 @@ export function checkEmulatorBuild(emulatorBuild: string): void {
 function isValidBoolean(value: string): boolean {
   return value === 'true' || value === 'false';
 }
+
+export function checkDiskSize(diskSize: string): void {
```
Comment (ychescale9) [inline src/input-validator.ts:73]: would be good to add some tests for this, but don't stress about it😃

**Review by ychescale9 — APPROVED** [review]
Thanks!

File: src/input-validator.ts (hunk 1 above)
Comment (ViliusSutkus89) [reply src/input-validator.ts:73]: Implemented tests uncovered a corner case, which is already handled :)
